### PR TITLE
IB-3286 Add /lists EP

### PR DESCRIPTION
**Pull request by sam-ludlow**
https://io1dev.atlassian.net/browse/IB-3286

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -6560,6 +6717,425 @@ components:
...
+                type: object
+                properties:
+                  uuid:
+                    description: 'Invotra UUID of the region.'
+                    type: string
+                    format: uuid
+                    example: '01234567-89ab-cdef-1234-56789abcdef0'
+          required:
+            - title
+            - body
```
Comment (ParryLuke) [inline api.swagger.yaml]: List body isn't mandatory for create/update

File: api.swagger.yaml [hunk 2]
```diff
@@ -6560,6 +6717,425 @@ components:
...
+            - 'content'
+            - 'site_section'
+            - 'link'
+        title:
+          description: 'Title of the content or link.'
+          type: string
+          example: 'More information here.'
+      required:
+      - type
+      - title
```
Comment (ParryLuke) [inline api.swagger.yaml]: title is optional for the list items

File: api.swagger.yaml [hunk 3]
```diff
@@ -6560,6 +6717,425 @@ components:
...
+      - $ref: '#/components/schemas/list-list-common-schema'
+      - type: object
+        properties:
+          type:
+            example: 'link'
+          content:
+            type: string
+            description: 'The URL of the link.'
+            example: 'https://example.com'
+
```
Comment (ParryLuke) [inline api.swagger.yaml]: remove empty line

File: api.swagger.yaml [hunk 4]
```diff
@@ -6560,6 +6717,425 @@ components:
           type: string
           example: '#FFFFFF'
       type: object
+
```
Comment (ParryLuke) [inline api.swagger.yaml]: empty line

File: api.swagger.yaml [hunk 5]
```diff
@@ -1682,6 +1682,163 @@ paths:
...
+      responses:
+        '200':
+          description: 'The list was successfully deleted.'
+        '400':
+          description: 'The supplied UUID was malformed.'
+        '403':
+          description: 'User is not authorized to delete this list.'
+        '404':
+          description: 'The UUID does not correspond to an existing list.'
+
```
Comment (ParryLuke) [inline api.swagger.yaml]: empty line

File: api.swagger.yaml [hunk 6]
```diff
@@ -1682,6 +1682,163 @@ paths:
         '422':
           description: JSON schema validation failure
       description: Report a post as the current user.
+
```
Comment (ParryLuke) [inline api.swagger.yaml]: empty line

File: api.swagger.yaml [hunk 7]
```diff
@@ -6560,6 +6717,425 @@ components:
...
+          example: Dorothy
+        surname:
+          description: The user's surname
+          type: string
+          example: Whitehead
+        user_avatar:
+          description: The user's image
+          type: string
+          example: "https://dev.invotra.com/sites/default/files/styles/large_avatar/public/Dorothy%20Whitehead.jpg?itok=6p19"
+
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml (hunk 1 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml (hunk 2 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml (hunk 3 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml (hunk 4 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml (hunk 5 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml (hunk 6 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml [hunk 8]
```diff
@@ -6560,6 +6715,420 @@ components:
...
+            example: 'content'
+          content:
+            type: object
+            properties:
+              uuid:
+                description: 'Invotra UUID of the content.'
+                type: string
+                format: uuid
+                example: '01234567-89ab-cdef-1234-56789abcdef0'
+    list-list-site_section-schema:
```
Comment (jmorahan) [inline api.swagger.yaml]:
```suggestion
    list-list-site-section-schema:
```

File: api.swagger.yaml [hunk 9]
```diff
@@ -6560,6 +6715,420 @@ components:
...
+          description: 'The review date timestamp for the list (in seconds).'
+          type: integer
+          example: 1510941596
+        list:
+          description: 'List of content and links in the list.'
+          type: array
+          items:
+            oneOf:
+              - $ref: '#/components/schemas/list-list-content-schema'
+              - $ref: '#/components/schemas/list-list-site_section-schema'
```
Comment (jmorahan) [inline api.swagger.yaml]:
```suggestion
              - $ref: '#/components/schemas/list-list-site-section-schema'
```

File: api.swagger.yaml [hunk 10]
```diff
@@ -6560,6 +6715,420 @@ components:
...
+              - $ref: '#/components/schemas/list-list-link-schema'
+            discriminator:
+              propertyName: type
+              mapping:
+                content: '#/components/schemas/list-list-content-schema'
+                site_section: '#/components/schemas/list-list-site_section-schema'
```
Comment (jmorahan) [inline api.swagger.yaml]:
```suggestion
                site_section: '#/components/schemas/list-list-site-section-schema'
```

File: api.swagger.yaml [hunk 11]
```diff
@@ -6560,6 +6715,420 @@ components:
...
+          description: The user's title
+          enum:
+            - ''
+            - Dr
+            - Mr
+            - Miss
+            - Mrs
+            - Ms
+          type: string
+          example: Mr
```
Comment (jmorahan) [inline api.swagger.yaml]:
```suggestion
          example: Ms
```
for consistency with the adjacent example firstname

File: api.swagger.yaml [hunk 12]
```diff
@@ -1682,6 +1682,161 @@ paths:
         '422':
           description: JSON schema validation failure
       description: Report a post as the current user.
+  '/lists':
+    post:
+      tags:
+        - 'lists'
+      description: 'Provision a new list.'
+      responses:
+        '201':
```
Comment (jmorahan) [inline api.swagger.yaml]: should we also return a response body with the uuid?

Comment (sam-ludlow) [reply api.swagger.yaml]: put in same way as https://io1dev.atlassian.net/browse/ITB-1136

**Comment by MireilleHelle** [comment]
Changes will be merged in https://github.com/invotra/api/pull/219/files instead.

**Pull request closed without merging**